### DISCUSSION TITLE
Lesson 12: How to write these Tests?

### DISCUSSION THREAD

**Discussion by Ifra001** [Cyfrin/foundry-full-course-cu#2617] [Q&A]
So I am at the Stablecoin section of the course and at the part where I have to finish writing the tests on my own. I was wondering how I could write these specific type of tests. If Patrick has taught them, please provide a timestamp or offer some areas where I can learn about them. 

Its these type of tests:

1. ```testRevertsIfMintFails()```
2. ```testRevertsIfTransferFails()```
3. ```testRevertsIfTransferFromFails()```

if you don't understand what I am talking about, please go to the official code repository and see these tests

**Comment by EngrPips** [comment]
Hello @DontMind-me, Looking at the official code repository should give you a hint on how to write the test.

**Reply by Ifra001** [reply]
I looked at them but don't quite understand them...

**Reply by EngrPips** [reply]
The main gist is you need to create a specific `ERC20` token that will fulfill the condition you want to test for each test and you use the `ERC20` token in the deployment of the protocol. Does this make sense to you?
